src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double SCALE_FACTOR = 1.5;
+    private int size;
+    private Object[] arrList;
 
+    public ArrayList() {
+        arrList = new Object[DEFAULT_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public void add(T value) {
+        resize();
+        arrList[size] = value;
+        size++;
     }
 
     @Override
-    public void addAll(List<T> list) {
-
+    public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index: " + index
+                    + " out of size " + size);
+        }
+        resize();
+        System.arraycopy(arrList, index, arrList, index + 1, size - index);
+        arrList[index] = value;
+        size++;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) arrList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        arrList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T value = (T) arrList[index];
+        size--;
+        System.arraycopy(arrList, index + 1, arrList, index, size - index);
+        return value;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (element == arrList[i] || element != null && element.equals(arrList[i])) {
+                T value = (T) arrList[i];
+                remove(i);
+                return value;
+            }
+        }
+        throw new NoSuchElementException("this element does not exist");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("index is not valid");
+        }
+    }
+
+    private void resize() {
```

Comment:
What is the difference between resize and grow? I mean word meaning.
Better remove resize() and use IF before call grow() where you need it